src/Concerns/RedactsHeaders.php

```
@@ -0,0 +1,73 @@
+use function trim;
+
+trait RedactsHeaders
+{
+    /**
+     * @param  list<string>  $keys
+     */
+    private function redactHeaders(HeaderBag $headers, array $keys = []): HeaderBag
+    {
+        foreach ($keys as $key) {
+            if (! $headers->has($key)) {
+                continue;
+            }
+
+            $headers->set($key, array_map(fn ($value) => match (strtolower($key)) {
+                'authorization', 'proxy-authorization' => $this->redactAuthorizationHeaderValue((string) $value), // @phpstan-ignore cast.string
+                'cookie' => $this->redactCookieHeaderValue((string) $value), // @phpstan-ignore cast.string
+                default => $this->redactHeaderValue((string) $value), // @phpstan-ignore cast.string
+            }, $headers->all($key)));
+        }
+
+        return $headers;
+    }
+
+    private function redactHeaderValue(string $value): string
+    {
+        return '['.strlen($value).' bytes redacted]';
+    }
+
+    private function redactAuthorizationHeaderValue(string $value): string
+    {
+        if (strpos($value, ' ') === false) {
+            return $this->redactHeaderValue($value);
+        }
+
+        [$type, $remainder] = explode(' ', $value, 2);
+
+        if (in_array(strtolower($type), ['basic', 'bearer', 'digest', 'hoba', 'mutual', 'negotiate', 'ntlm', 'vapid', 'scram', 'aws4-hmac-sha256'], true)) {
+            return $type.' '.$this->redactHeaderValue($remainder);
+        }
+
+        return $this->redactHeaderValue($value);
+    }
+
+    private function redactCookieHeaderValue(string $value): string
+    {
+        $cookies = explode(';', $value);
+
+        try {
+            return implode('; ', array_map(function ($cookie) {
+                [$name, $value] = explode('=', $cookie, 2);
+
+                return trim($name).'='.$this->redactHeaderValue($value);
+            }, $cookies));
+        } catch (Throwable) {
+            return $this->redactHeaderValue($value);
+        }
```

🍸 should we report the exception?